changes/6283.dependencies

```diff
@@ -0,0 +1 @@
+Updated `mkdocs-material` to `~9.5.39`
```

Comment:
```suggestion
Updated documentation dependency `mkdocs-material` to `~9.5.39`.
```

Also, we haven't always been consistent about this but I've tried to assert that development-only dependency updates go in `.housekeeping` and only "real" dependencies go in `.dependencies`.

Reply:
Yeah sorry about that. I wonder if there is a sure-fire way to tell development-only dependencies from other dependencies?

Reply:
Inspection of `pyproject.toml` mostly :-/